training-1/src/main.rs

```diff
@@ -126,8 +131,11 @@ fn main() {
         }
 
         if !level_complete {
-            if let Some(cmd) = vm.console.try_process_command() {
-                vm.execute_command(cmd);
+            if let Some(cmd) = vm.console.get_next_command() {
```

Comment:
That looks like a really good idea to be honest. I'm wary about doing this all in one PR though and having it get outside of the scope.

Perhaps its best if we merge this one with the new `CommandSystem`, then we'll open another one which involves restructuring the code?